tests/hypothesis_strategies/constraints.py

```diff
@@ -71,6 +74,106 @@ def discrete_excludes_constraints(
     return DiscreteExcludeConstraint(parameter_names, conditions, combiner)
 
 
+@st.composite
+def discrete_dependencies_constraints(
```

Comment:
they are documented in the respective docstring of the constraints

Here the tldr:
- `parameters` is a list of prameters that each causes a list of other parameters to become irrelevant if a certain condition is met (for each respective entry in parameters)
- `affected_parameters` is that other list of parameters, its a list of lists where each sub-list are all the parameters whose values become irrelevant (degenerate) when the condition for the respective entry in `parameters` is met. conditions dont appear here in the strategy as they are auto-generated, but that leads to various possible problems for affected_parameters. We might need to complciate this even further and also pass conditions (because currently the conditions dont necessarily make sense) but imo thats best done in a PR that actually uses constraint strats

Reply:
Ok, makes sense to have the description just at a single point :)